### PR TITLE
Provide documentation for Solace PubSub+ Event Broker scaler

### PR DESCRIPTION
**Pull request by dennis-brinley**
Provide documentation for Solace PubSub+ Event Broker scaler

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)
- [x] A PR is opened to update the documentation on [our docs repo](https://github.com/kedacore/keda-docs)

Fixes #

### CODE REVIEW

File: content/docs/2.3/scalers/solace.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
++++
+title = "Solace PubSub+"
+availability = "2.3+"
```
Comment (tomkerkhove) [inline content/docs/2.3/scalers/solace.md:3]: We have already shipped v2.3, can you please update this and move to the correct folder please?
```suggestion
availability = "2.4+"
```

Comment (tomkerkhove) [reply content/docs/2.3/scalers/solace.md:3]: Actually it is already in v2.4 so let's delete this file.

File: content/docs/2.4/scalers/solace.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
++++
+title = "Solace PubSub+"
+availability = "2.3+"
```
Comment (tomkerkhove) [inline content/docs/2.4/scalers/solace.md:3]:
```suggestion
availability = "2.4+"
```

File: content/docs/2.4/scalers/solace.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+maintainer = "Community"
+description = "Scale applications based on Solace PubSub+ Broker Queues"
+layout = "scaler"
+go_file = "solace_scaler"
++++
+
+### Trigger Specification
+This specification describes the `solace-queue` trigger that scales based on a Solace PubSub+ queues.
```
Comment (tomkerkhove) [inline content/docs/2.4/scalers/solace.md:11]: The scaler is called Pub/Sub, but here we refer to queues, I think it's best to align the naming.

What do you think @zroubalik?

File: content/docs/2.4/scalers/solace.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    msgSpoolUsageTarget:  '10000'
+    username:             semp-user
+    password:             semp-pwd
+    usernameEnv:          ENV_VAR_USER
+    passwordEnv:          ENV_VAR_PWD
+```
+
+**Parameter list:**
+- `solaceSempBaseURL` - Solace SEMP Endpoint in format: `<protocol>://<host-or-service>:<port>`
+- `msgVpn` - Message VPN hosted on the Solace broker
```
Comment (tomkerkhove) [inline content/docs/2.4/scalers/solace.md:30]: We typically don't abbreviate things
```suggestion
- `messageVpn` - Message VPN hosted on the Solace broker
```

File: content/docs/2.4/scalers/solace.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+- `queueName` - Message Queue to be monitored
+- `msgCountTarget` - The target number of messages manageable by a pod. The scaler will cause the replicas to increase if the queue backlog is greater than that value per active replica.
```
Comment (tomkerkhove) [inline content/docs/2.4/scalers/solace.md:32]: We typically don't abbreviate things

```suggestion
- `messageCountTarget` - The target number of messages manageable by a pod. The scaler will cause the replicas to increase if the queue backlog is greater than that value per active replica.
```

File: content/docs/2.4/scalers/solace.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+- `msgSpoolUsageTarget` - The target spool usage manageable by a pod. The scaler will cause the
```
Comment (tomkerkhove) [inline content/docs/2.4/scalers/solace.md:33]: We typically don't abbreviate things

```suggestion
- `messageSpoolUsageTarget` - The target spool usage manageable by a pod. The scaler will cause the 
```

File: content/docs/2.4/scalers/solace.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+- `passwordEnv` - Environment variable set with password for the user account
+
+**Parameter Requirements:**
+- Parameters resolving the target queue are all **required:** `solaceSempBaseURL`, `msgVpn`, `queueName`
+- **At least** one of `msgCountTarget` or `msgSpoolUsageTarget` is **required.** If both values are present, the metric value resulting in the higher desired replicas will be used. (Standard KEDA/HPA behavior)
+- `username` and `password` are **required** for the `solace-queue` trigger to function. However, these values may be passed using different methods. See [Authentication Parameters](#authentication-parameters) below.
+
+### Authentication Parameters
+The Solace PubSub+ Scaler polls the SEMP REST API to monitor target queues. Currently, the scaler only supports basic authentication. Username and Password credentials are both required. Credentials may be passed using one of three different methods:
+- Using a `TriggerAuthentication` CRD to configure the `username` and `password`. The `TriggerAuthentication` must reference a `Secret` with the credentials defined.
```
Comment (tomkerkhove) [inline content/docs/2.4/scalers/solace.md:46]: Can you elaborate why a secret is required? In theory people can bring the value from wherever they want as long as it uses the correct parameter name.

File: content/docs/2.4/scalers/solace.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+- Environment Variables - Set fields `usernameEnv` and `passwordEnv` in the trigger configuration to the appropriate environment variables
+- Clear Text - Set fields `username` and `password` in the trigger configuration in clear text
```
Comment (tomkerkhove) [inline content/docs/2.4/scalers/solace.md:48]: This section is typically only used for TriggerAuthentication
```suggestion
```

File: content/docs/2.4/scalers/solace.md (hunk 7 above)
Comment (tomkerkhove) [reply content/docs/2.4/scalers/solace.md:46]: Can you align with this approach please?
https://keda.sh/docs/2.3/scalers/azure-service-bus/#authentication-parameters

File: content/docs/2.3/scalers/solace.md (hunk 1 above)
Comment (dennis-brinley) [reply content/docs/2.3/scalers/solace.md]: Done

File: content/docs/2.4/scalers/solace.md (hunk 2 above)
Comment (dennis-brinley) [reply content/docs/2.4/scalers/solace.md]: Done

File: content/docs/2.4/scalers/solace.md (hunk 4 above)
Comment (dennis-brinley) [reply content/docs/2.4/scalers/solace.md:30]: Done - Corresponding code change pending

File: content/docs/2.4/scalers/solace.md (hunk 5 above)
Comment (dennis-brinley) [reply content/docs/2.4/scalers/solace.md]: Done - Corresponding code change pending

File: content/docs/2.4/scalers/solace.md (hunk 6 above)
Comment (dennis-brinley) [reply content/docs/2.4/scalers/solace.md]: Done - Corresponding code change pending

File: content/docs/2.4/scalers/solace.md (hunk 7 above)
Comment (dennis-brinley) [reply content/docs/2.4/scalers/solace.md:46]: Striking the reference to 'Secret'

File: content/docs/2.4/scalers/solace.md (hunk 3 above)
Comment (dennis-brinley) [reply content/docs/2.4/scalers/solace.md:11]: I changed the doc title="Solace PubSub+ Queue" -- The trigger ID is still "solace-queue". Let me know if this is acceptable.
**_There is no consistency in the existing scalers. e.g.:_**
"Rabbit MQ Queue" -> rabbitmq
"ActiveMQ Artemis" -> artemis-queue
"IBM MQ" -> ibmmq
Some reference company, product, technology, and scaler object type. But none do all of those things consistently between the scaler name and the scaler identifier.

Comment (dennis-brinley) [reply content/docs/2.4/scalers/solace.md:11]: Updated to scaler/trigger ID to "solace-event-queue" based on internal feedback. Doc title is also updated

File: content/docs/2.4/scalers/solace.md (hunk 8 above)
Comment (dennis-brinley) [reply content/docs/2.4/scalers/solace.md:48]: Updated

**Review by zroubalik — CHANGES_REQUESTED** [review]
Looking good, minor change needed only

**Review by tomkerkhove — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]

**Pull request merged**